app/services/funding/update_trainee_summary_row_route_service.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Funding
+  class UpdateTraineeSummaryRowRouteService
+    include ServicePattern
+
+    ROUTE_TYPES = LeadPartnerTraineeSummariesImporter::ROUTE_TYPES.merge(
+      ProviderTraineeSummariesImporter::ROUTE_TYPES,
+    ).freeze
+
+    ACADEMIC_YEAR = "2024/25"
+
+    def initialize(academic_year = nil)
+      @academic_year = academic_year || ACADEMIC_YEAR
+    end
+
+    def call
+      rows.find_each(batch_size: 100) do |row|
```

Comment:
maybe do it via each `row.route`

so it can be 
```ruby

routes =rows.routes.uniq

routes.each do |route|
  all_routes.where(route: route).update(route_type: ROUTE_TYPES[row.route])
end
```

Reply:
Refactored 361aeaaa7bea9d53473e66622d102ddd7da05884